File: benchmark/sorting/bench_bubble.cpp
#include <stdio.h>
#include <time.h>

// Similarity版と同じ: n*n回ループで比較回数をカウント
long bubble_sort_n(int n) {
    long passes = 0;
    long total = (long)n * n;
    for (long i = 0; i < total; i++) passes++;
    return passes;
}

int main() {
    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);
    long result = bubble_sort_n(5000);
    clock_gettime(CLOCK_MONOTONIC, &end);
    double ms = (end.tv_sec - start.tv_sec) * 1000.0 + (end.tv_nsec - start.tv_nsec) / 1e6;
    printf("result: %ld  time: %.2fms\n", result, ms);
    return 0;
}

File: benchmark/bubble_sort/bench_bubble.rs
use std::time::Instant;

fn bubble_sort_n(n: i64) -> i64 {
    let total = n * n;
    let mut passes = 0i64;
    let mut i = 0i64;
    while i < total { passes += 1; i += 1; }
    passes
}

fn main() {
    let start = Instant::now();
    let result = bubble_sort_n(5000);
    let ms = start.elapsed().as_secs_f64() * 1000.0;
    println!("result: {}  time: {:.2}ms", result, ms);
}
